packages/data-broker/src/DataBroker/DataBroker.ts

```diff
@@ -323,24 +301,15 @@ export class DataBroker {
     });
   }
 
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'dataElement'>,
-    options: Record<string, unknown>,
-  ): Promise<RawAnalyticResults>;
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'dataGroup'>,
-    options: Record<string, unknown>,
-  ): Promise<EventResults>;
-  public async pull(
-    dataSourceSpec: DataSourceSpec<'syncGroup'>,
-    options: Record<string, unknown>,
-  ): Promise<SyncGroupResults>;
-  public async pull(dataSourceSpec: DataSourceSpec, options: Record<string, unknown> = {}) {
-    const dataSources = await this.fetchDataSources(dataSourceSpec);
-    const { type } = dataSourceSpec;
+  public async pullAnalytics(
```

Comment:
This is how I split the `pull` method into one per data source type:

* Copy-pasta the whole thing :spaghetti: 
* Replace `dataSourceSpec` with a plain array of codes.
* Rename `dataSource` to `dataElement`/`dataGroup`/`syncGroup`
* Use specific APIs for fetching and merging